### PR TITLE
Proposal/9/convert replacement

### PR DESCRIPTION
**Pull request by chfsx**
As described in #5172, we want to reduce the usage of php's `exec` to a minimum. zip/unzip, for example, have already been replaced. this PR here now replaces `convert` commands in the same way.

in the code-base as many places as possible are already refactored, where I unfortunately could not refactor was in the T&A, because there `convert` is partly also used for building image maps, which is not supported by the new service (yet).

The actual service is committed here: https://github.com/ILIAS-eLearning/ILIAS/commit/622c97bb53b8dc6035bb1d5a70cb47867d38a3f8 which is Unit-tested as well

The following maintainers can view a summary of their relevant changes:

@alex40724 
- https://github.com/ILIAS-eLearning/ILIAS/commit/165e4b9058903bc5dfaacc7189535175812a2590: Modules/Blog
- https://github.com/ILIAS-eLearning/ILIAS/commit/4b708403efe102ef14b2708dc5881fa1ab05604c: Services/MediaObjects
- https://github.com/ILIAS-eLearning/ILIAS/commit/1ab3295d419a624a2cc1402e503c2938333edbc1: Services/Style (Content)

@smeyer-ilias 
- https://github.com/ILIAS-eLearning/ILIAS/commit/14947d164bfca88be1caae31dc63dccb44d4ab99: Modules/Poll

@mjansenDatabay 
- https://github.com/ILIAS-eLearning/ILIAS/commit/c6b6213730bb2a98307ceeff95bd11580c16258c: Services/Certificate

@kergomard 
- https://github.com/ILIAS-eLearning/ILIAS/commit/55c2f2e44dc9e769790242c46c5a732c6f9e2ae6: Services/Object

The old methods are marked here as deprecated for ILIAS 10: https://github.com/ILIAS-eLearning/ILIAS/commit/4ebc668917d41cc77f94022fb3ef9c66c0426ae3

But I encourage everyone to convert files to the IRSS anyway, so that they can also use flavours, for example, which are an easy way to manage thumbnails and different sizes, etc.

### CODE REVIEW

**Comment by chfsx** [comment]
@kergomard of course, the change in the tile images is only relevant if you don't make the switch to the IRSS anyway and can thus use the flavours directly, which is of course much easier. I could drop your commit in that case if needed

File: Services/Certificate/classes/Helper/ilCertificateUtilHelper.php [hunk 1]
```diff
@@ -42,11 +53,15 @@ public function prepareFormOutput(string $string): string
...
-        string $backgroundColor = ''
+        string $geometry = ''
     ): void {
-        ilShellUtil::convertImage($from, $to, $targetFormat, $geometry, $backgroundColor);
+        $this->image_converter->convertToFormat(
+            $from,
+            $to,
+            ImageOutputOptions::FORMAT_JPG,
+            $geometry === '' ? null : (int)$geometry,
+            $geometry === '' ? null : (int)$geometry,
```
Comment (mjansenDatabay) [inline Services/Certificate/classes/Helper/ilCertificateUtilHelper.php]:
```suggestion
            $geometry === '' ? null : (int) $geometry,
            $geometry === '' ? null : (int) $geometry,
```

**Review by mjansenDatabay — APPROVED** [review]
Thx @chfsx !

**Review by kergomard — APPROVED** [review]
Thank you very much @chfsx !
The code for the `TileImages` will be gone with ILIAS 9 in favor of the IRSS, feel free to merge this and then I work on from there.

**Review by alex40724 — APPROVED** [review]
Great! Thanks.

**Comment by chfsx** [comment]
@smeyer-ilias any Feedback for https://github.com/ILIAS-eLearning/ILIAS/commit/14947d164bfca88be1caae31dc63dccb44d4ab99 ? I'd like to announce and merge this next JF on monday.

**Comment by matthiaskunkel** [comment]
Jour Fixe, 26 JUN 2023: We highly appreciate this suggestion and accept the PR for trunk. Please merge the PR when Stefan has given his feedback to this PR.

**Comment by chfsx** [comment]
HI @smeyer-ilias could you please give your feedback for commit https://github.com/ILIAS-eLearning/ILIAS/commit/14947d164bfca88be1caae31dc63dccb44d4ab99 this week so i can merge it? thank you!

**Comment by chfsx** [comment]
HI @smeyer-ilias could you please give your feedback for commit https://github.com/ILIAS-eLearning/ILIAS/commit/14947d164bfca88be1caae31dc63dccb44d4ab99, I'd like to merge this today.

**Review by smeyer-ilias — APPROVED** [review]
Thanks, for this improvement

**Pull request merged**